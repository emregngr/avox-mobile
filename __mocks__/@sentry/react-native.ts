import type { ComponentType } from 'react'

const mockedSentry = {
  init: jest.fn(),
  wrap: <T extends ComponentType<any>>(Component: T): T => Component,
  withScope: jest.fn((callback: (scope: any) => void) =>
    callback({
      setTag: jest.fn(),
      setLevel: jest.fn(),
      setUser: jest.fn(),
      setContext: jest.fn(),
    }),
  ),
  captureException: jest.fn(),
  captureMessage: jest.fn(),
  addBreadcrumb: jest.fn(),
  configureScope: jest.fn(),
  setUser: jest.fn(),
  setTag: jest.fn(),
  setContext: jest.fn(),
  setRelease: jest.fn(),
  setDist: jest.fn(),
  setEnvironment: jest.fn(),
  close: jest.fn(() => Promise.resolve(true)),
  flush: jest.fn(() => Promise.resolve(true)),
}

module.exports = mockedSentry

export type mockedSentryType = typeof mockedSentry
