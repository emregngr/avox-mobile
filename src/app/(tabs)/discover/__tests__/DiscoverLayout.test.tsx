import { render } from '@testing-library/react-native'
import React from 'react'

import DiscoverLayout from '@/app/(tabs)/discover/_layout'

describe('DiscoverLayout', () => {
  it('should render the Slot component successfully', () => {
    render(<DiscoverLayout />)
  })
})

describe('DiscoverLayout Snapshot', () => {
  it('should render the DiscoverLayout successfully', () => {
    const { toJSON } = render(<DiscoverLayout />)

    expect(toJSON()).toMatchSnapshot()
  })
})
