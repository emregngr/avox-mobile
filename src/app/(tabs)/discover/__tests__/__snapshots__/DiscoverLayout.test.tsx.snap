// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`DiscoverLayout Snapshot should render the DiscoverLayout successfully 1`] = `null`;
